crates/teepot/src/config/mod.rs

```diff
@@ -269,15 +269,30 @@ fn init_telemetry(
         )
         .build();
 
-    // Configure the OTLP exporter
+    // Parse the protocol from the configuration
+    let protocol = protocol_from_string(&config.otlp.protocol)?;
+
+    // Configure the OTLP exporter based on the protocol
+    let exporter_builder = opentelemetry_otlp::LogExporter::builder();
+
+    // Choose transport based on protocol
+    let exporter = if protocol == opentelemetry_otlp::Protocol::Grpc {
+        exporter_builder
+            .with_tonic()
+            .with_endpoint(&config.otlp.endpoint)
+            .with_protocol(protocol)
+            .build()?
+    } else {
+        exporter_builder
+            .with_http()
+            .with_endpoint(&config.otlp.endpoint)
+            .with_protocol(protocol)
+            .build()?
+    };
```

Comment:
Prefer match to future proof against new protocols being added not ending up incorrectly implemented through http; the enum isn't marked as non exhaustive so this isn't strictly required, but it's the kind of thing that's easy to miss when doing reviews for the impacts of breaking changes (from experience)

Reply:
you are totally right.. here we go